cpp/examples/build.sh

```diff
@@ -20,7 +20,7 @@ build_example() {
   build_dir="${example_dir}/build"
 
   # Configure
-  cmake -S ${example_dir} -B ${build_dir} -Dcudf_ROOT="${LIB_BUILD_DIR}"
+  cmake -S ${example_dir} -B ${build_dir} -Dcudf_ROOT="${LIB_BUILD_DIR}" -DCUDF_TAG=branch-24.06
```

Comment:
Since this file has a hardcoded version number, you have to update `ci/release/update-version.sh` so it stays in sync. You probably need to edit the line that is currently in there, since `fetch_dependencies.cmake` no longer has the version number:

https://github.com/rapidsai/cudf/blob/79d2dba141e0cbc464810a0b214d2c714ebe1a18/ci/release/update-version.sh#L72-L73

Reply:
Put the `CUDF_TAG` back in `fetch_dependencies.cmake` and imported it in `set_cuda_architecture.cmake` for consistency.